xapian-letor/tests/api_letor.cc

```diff
@@ -33,32 +31,28 @@
 
 using namespace std;
 
-DEFINE_TESTCASE(assign_db_path1, !backend) {
-   Xapian::Ranker *ranker = new Xapian::ListNETRanker();
-   ranker->set_database_path("db_path");
-   TEST_EQUAL("db_path", ranker->get_database_path());
-   delete ranker;
-   ranker = new Xapian::SVMRanker();
-   ranker->set_database_path("db_path");
-   TEST_EQUAL("db_path", ranker->get_database_path());
-   delete ranker;
+DEFINE_TESTCASE(check_db_path, !backend) {
+   Xapian::ListNETRanker ranker1;
+   ranker1.set_database_path("db_path");
+   TEST_EQUAL("db_path", ranker1.get_database_path());
+   Xapian::SVMRanker ranker2;
+   ranker2.set_database_path("db_path");
+   TEST_EQUAL("db_path", ranker2.get_database_path());
```

Comment:
`ranker1` and `ranker2` is clumsy.  You could just wrap each ranker's part of the test in a block, and call them `ranker`.